src/middlewared/middlewared/plugins/truenas_connect/update.py

```diff
@@ -79,17 +79,39 @@ async def config_extend(self, config):
             config['certificate'] = config['certificate']['id']
         return config
 
+    @private
+    async def ha_vips(self):
+        vips = []
+        for interface in await self.middleware.call('interface.query'):
+            for vip_entry in interface.get('failover_virtual_aliases', []):
+                vips.append(vip_entry['address'])
+        return vips
+
     @private
     async def validate_data(self, old_config, data):
         verrors = ValidationErrors()
-
-        # Ensure at least one interface or at least one IP is provided (unless use_all_interfaces is True)
-        if data['enabled'] and not data['ips'] and not data['interfaces'] and not data['use_all_interfaces']:
-            verrors.add(
-                'tn_connect_update',
-                'At least one IP or interface must be provided when TrueNAS Connect is enabled '
-                '(or use_all_interfaces must be set to true)'
-            )
+        if data['enabled']:
+            if await self.middleware.call('system.is_ha_capable'):
+                # In case of HA, we want to ensure following:
+                # 1) We have VIP available
+                # 2) User has not specified interfaces/use all interfaces
+                if not await self.ha_vips():
+                    verrors.add(
+                        'tn_connect_update.enabled',
+                        'HA systems must be in a healthy state to enable TNC ensuring we have VIP available'
+                    )
+                for k in filter(lambda k: data[k], ('interfaces', 'use_all_interfaces')):
```

Comment:
Will we run into any issue with use_all_interfaces as on by default?

Reply:
HA systems are meant to be used using VIP,  @yocalebo do you think we should allow specific or all interfaces ?
Basically these attrs control what IPs we send to TNC based on what is configured on selected/all interfaces.

For public or special ip cases, we are allowing that right now separately